src/clients/ETHClient.ts

```diff
@@ -91,15 +92,16 @@ export class ETHClient {
   private constructor(
     public readonly configProvider: NetworkConfigProvider,
     public readonly blockchain: typeof ETHClient.SUPPORTED_BLOCKCHAINS[number],
-    public readonly tokenClient: TokenClient
+    public readonly tokenClient: TokenClient,
+    public readonly rpcURL: string,
   ) { }
 
   public static instance(opts: ETHClientOpts) {
-    const { configProvider, blockchain, tokenClient } = opts;
+    const { configProvider, blockchain, tokenClient, rpcURL } = opts;
 
     if (!ETHClient.SUPPORTED_BLOCKCHAINS.includes(blockchain)) throw new Error(`unsupported blockchain - ${blockchain}`);
```

Comment:
Think this can be changed to:
```typescript
if (!BlockchainUtils.isEvmChain(blockchain) || blockchain === "Carbon") {
  throw new Error(`unsupported blockchain - ${blockchain}`);
}
```

Then after that, you can remove these 3 variables
![Screenshot 2025-03-26 132540](https://github.com/user-attachments/assets/be1477d0-876d-4ac5-9802-dc9e8b83ee89)